### PR TITLE
Update some of the example tests to use assert_that

### PR DESCRIPTION
**Pull request by aaltay**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aaltay** [comment]
R: @robertwb Please review.

File: sdks/python/apache_beam/transforms/util.py [hunk 1]
```diff
@@ -196,6 +199,16 @@ def _empty(actual):
   return _empty
 
 
+def contains_in_any_order(expected):
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/util.py]: Is the only difference between this and equal_to is that this requires countable elements and equal_to requires comparable elements? Seems a subtle distinction...

**Comment by robertwb** [comment]
Much nicer, thanks! LGTM

File: sdks/python/apache_beam/transforms/util.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/transforms/util.py]: Good point. I think this is redundant. I will do more cleanups of the examples, I will remove this function together with those changes.

**Comment by aaltay** [comment]
Thank you.

**Comment by silviulica** [comment]
LGTM - Ahmet get this in please as soon as you can.

**Pull request closed without merging**